blueocean-bitbucket-pipeline/src/main/java/io/jenkins/blueocean/blueocean_bitbucket_pipeline/BitbucketApi.java

```diff
@@ -0,0 +1,63 @@
+package io.jenkins.blueocean.blueocean_bitbucket_pipeline;
+
+import com.cloudbees.plugins.credentials.common.StandardUsernamePasswordCredentials;
+import io.jenkins.blueocean.blueocean_bitbucket_pipeline.model.BbBranch;
+import io.jenkins.blueocean.blueocean_bitbucket_pipeline.model.BbPage;
+import io.jenkins.blueocean.blueocean_bitbucket_pipeline.model.BbProject;
+import io.jenkins.blueocean.blueocean_bitbucket_pipeline.model.BbRepo;
+import io.jenkins.blueocean.blueocean_bitbucket_pipeline.model.BbSaveContentResponse;
+import io.jenkins.blueocean.blueocean_bitbucket_pipeline.model.BbUser;
+import org.antlr.v4.runtime.misc.NotNull;
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.NoExternalUse;
+
+import javax.annotation.CheckForNull;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.Map;
+
+/**
+ * @author Vivek Pandey
+ */
+@Restricted(NoExternalUse.class)
+public abstract class BitbucketApi {
```

Comment:
It's strictly internal use with only two internal implementation already in place, but I will add some docs.